core-httpclient-impl/src/main/java/com/optimizely/ab/OptimizelyFactory.java

```diff
@@ -48,6 +49,14 @@
  */
 public final class OptimizelyFactory {
     private static final Logger logger = LoggerFactory.getLogger(OptimizelyFactory.class);
+    private static CloseableHttpClient customHttpClient;
+
+    public static void setCustomHttpClient(CloseableHttpClient httpClient) {
+        if (httpClient == null) {
+            logger.error("CloseableHttpClient must not be null. Please initialize object.");
+        }
+        customHttpClient = httpClient;
+    }
```

Comment:
I think we can remove this method (not consistent with the rest of set methods here for property settings) and feed into newInstance parameter.